Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+
+        Notes
+        -----
+        Overrides the pandas append functionality!
+        """
+        if not isinstance(other, pd.DataFrame):
+            other = self._constructor(data=[other],
+                                      columns=[c for c in self.columns if c not in self._INTERNAL_COLUMN_NAMES],
+                                      index=[0])
+        other.index = self._new_id(len(other), force_list=True)
+        if self._WEIGHTS_COLUMN not in other.columns:
+            other[self._WEIGHTS_COLUMN] = 1
+
+        # coerce incompatibilities: this happens when appending a list
+        #  - category dtypes must match, coerce them
+        for i, column in enumerate(self.columns):
+            if self.dtypes[i].name == 'category':
+                new_cats = [v for v in set(other[column]) if v not in self[column].cat.categories and v not in Variable.MISSING_VALUES]
+                self[column] = self[column].cat.add_categories(new_cats)
+                self.domain[column].values += new_cats
+                other[column] = other[column].astype('category',
+                                                     categories=self[column].cat.categories,
+                                                     ordered=self[column].cat.ordered)
+        # also coerce into the proper class
+        result = self.from_dataframe(self.domain, super().append(other))
+        # append doesn't transfer properties for some reason
+        result._transfer_properties(self)
+        return result
+
+    @classmethod
+    def concatenate(cls, tables, axis=1, reindex=True, colstack=True, rowstack=False):
+        """Concatenate tables by rows or columns.
+
+        The resulting table will always retain the properties (name etc.) of the first table in the sequence.
+
+        Parameters
+        ----------
+        tables : list[TableBase]
+            A list of tables to concatenate.
+        axis : int, optional, default 1
+            The axis by which to concatenate.
+            Axis 0 are rows, axis 1 are columns.
+        reindex : bool, optional, default True
+            Whether to generate a new index for the resulting Table.
+            If reindex is False
+             - when concatenating rows: some rows may have the same index
+             - when concatenating columns: the index of the first table is preserved
+        colstack : bool, optional, default True
+            Whether to stack columns when concatenating by columns.
+            No two columns may have the same name.
+            If colstack is True, the number of rows must match on all tables.
+            If colstack is False, perform an outer join instead.
+        rowstack : bool, optional, default False
+            Whether to stack rows when concatenating by rows.
+            If rowstack is True, the number of columns must match on all tables.
+            If rowstack is False, perform an outer join on the columns with NA values
+            in places without data.
```

Comment:
Inner joins and stacking isn't the same here. We could have only one `stacking` instead of two, but I don't think `join='inner'` would be the most correct way.